### DISCUSSION TITLE
IDLE loos on windows

### DISCUSSION THREAD

**Discussion by krb0095** [GrossfieldLab/loos#109] [General]
Hey All, 

I don't know if this would be helpful for your documentation, but I have been playing around with LOOS on windows. I was trying to get a IDLE to work for development of analysis codes, but installation of most of these programs do not have the power to access the windows Linux subsystem to have access to loos/pyloos. The two ways I found around it are:

1.  juypter notebooks/lab can run a sever over the WLS but often is sluggish in loading and working with trajectories/memory
2. The open source Visual studio code natively has access to the WLS and, using the conda environment, can be used to write and test loos code. 

Again not sure this is useful   

Kyle

**Comment by lgsmith** [comment]
I am not (and hope never to be again) a windows user, so I'm kind of
surprised you're trying to do this, but I do think that the point about
VSCode is interesting and likely helpful to people. Code is my editor of
choice for most things, so knowing that it behaves well in the windows
environment with their linux subsystem is probably great news for those who
were considering doing this.

Are you making this work by installing code in the linux subsystem, then
calling it from the linux terminal on the files or project directories you
are working with?

**Reply by krb0095** [reply]
I only use windows when I have to work away from my workstation.  The way I got it to work is by 

1. Setting up the WLS and installing LOOS via conda in linux 
2. downloading Code in the widows OS 
3. download the Visual Studio Code WLS  extension 
4. After the extension is installed the editor will sense the WLS and you can launch the code in that environment
5. use the terminal in Code to activate the conda environment

**Comment by Poruthoor** [comment]
Hi Kyle,

I just wanted to give my two cents. I'm a long-term window user myself primarily due to better Steam integration. I have been using loos inside Windows using similar routes you mentioned.

1. There are [multiple methods](https://stackoverflow.com/a/58068850) from which you can make conda work inside jupyter
2. I currently use VScode - WSL to write loos codes, just like you mentioned, as I migrated from Vim to VScode recently
3. I use WSL and Ubuntu LTS apps available for Windows for running loos codes from the terminal. It has been smooth. I would recommend the latter, and here is a [step-by-step guide](https://www.ssl.com/how-to/enable-linux-subsystem-install-ubuntu-windows-10/) that has worked for me and new members in the lab who uses Windows. Of course, you can do the same from the VSCode terminal if you want to stay on your editor.
